pilot/pkg/xds/workloadentry.go

```diff
@@ -145,9 +145,6 @@ func (sg *InternalGen) QueueUnregisterWorkload(proxy *model.Proxy) {
 // UpdateWorkloadEntryHealth updates the associated WorkloadEntries health status
 // based on the corresponding health check performed by istio-agent.
 func (sg *InternalGen) UpdateWorkloadEntryHealth(proxy *model.Proxy, event HealthEvent) {
-	if !features.WorkloadEntryHealthChecks {
```

Comment:
why is this removed?

Reply:
It is moved before calling this function to prevent unnecessary HealthEvent convert